src/jv_parse.c

```diff
@@ -649,7 +649,7 @@ static int stream_is_top_num(struct jv_parser* p) {
 static pfunc scan_line_comment(struct jv_parser* p, char ch, jv* out) {
   if(ch == '\n')
     p->scan = scan_json;
-  return OK;
+  return 0;
```

Comment:
Can we have some commentary in the commit message about why this change?

Reply:
Yeap can do that. When i rebased i tried to keep the existing commits as intact as possible as a first step.

Trying to understand the `0` -> `OK` change. `OK` is `static const presult OK = "output produced";` so i guess something else must have changed before the change? only reference i can find is in https://github.com/jqlang/jq/pull/2548#issuecomment-1628024506. Will digg more later.

Reply:
Yeah, I see that, but I'm not sure why that's needed, and anyways, there's places where we look for `OK` that need to be changed correspondingly.

Reply:
iirc the reason for this is fairly straightforward, which is that in the below block of jv_parse.c:

```
jv jv_parser_next(struct jv_parser* p) {
...
  while (!msg && p->curr_buf_pos < p->curr_buf_length) {
    ...
    msg = scan(p, ch, &value);
  }
```

the desired behavior, when a comment is encountered, is to simply ignore it and keep on going. Therefore the `while` loop must keep on going, which requires that `!msg` is true. Any other result will break the while loop, which would be incorrect.

That said, given that the return type of `scan()` is a pointer, it might be better for the `scan_xxx()` functions to all return NULL where they currently are returning of 0.

Reply:
@liquidaty thanks for the explanation. think i tried to return `OK` instead when played around a bit with the code and pretty sure it also worked for some reason, was confusing

Reply:
Maybe it would be useful to define `IGNORE_AND_CONTINUE` as a `presult` equal to NULL (or alternatively, not null, with other corresponding changes e.g. `!msg` becomes `msg == IGNORE_AND_CONTINUE`)